### PR TITLE
Capture telemetry data for Azure Spinnaker usage

### PR DESCRIPTION
**Pull request by milismsft**

### CODE REVIEW

**Comment by milismsft** [comment]
@alvadb @jinglouMSFT @scotmoor  PTAL

**Comment by milismsft** [comment]
- @huanhuansunmsft

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy [hunk 1]
```diff
@@ -65,6 +68,22 @@ public class AzureComputeClient extends AzureBaseClient {
...
+    computeClient.getClientInterceptors().add(new Interceptor() {
+      @Override
+      Response intercept(Interceptor.Chain chain) throws IOException {
+        Request.Builder builder = chain.request().newBuilder()
+        def oldHeaderValue = chain.request().header("User-Agent")
+        if (oldHeaderValue) {
+          builder.header("User-Agent", "AzureSpinnaker ${oldHeaderValue}")
+        } else {
+          builder.header("User-Agent", "AzureSpinnaker")
+        }
```
Comment (HuanhuanSunMSFT) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy]: I would simplify like this:
userAgentValue = oldHeaderValue ? "AzureSpinaker" : "AzureSpinnaker ${oldHeaderValue}";
builder.header("User-Agent", userAgentValue);

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy [hunk 2]
```diff
@@ -789,6 +789,13 @@ class AzureNetworkClient extends AzureBaseClient {
       Response intercept(Interceptor.Chain chain) throws IOException {
         Request.Builder builder = chain.request().newBuilder()
         builder.header("x-ms-client-request-id", UUID.randomUUID().toString())
+        // Add Azure Spinnaker telemetry capturing
+        def oldHeaderValue = chain.request().header("User-Agent")
+        if (oldHeaderValue) {
+          builder.header("User-Agent", "AzureSpinnaker ${oldHeaderValue}")
+        } else {
+          builder.header("User-Agent", "AzureSpinnaker")
+        }
```
Comment (HuanhuanSunMSFT) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy]: Same here.

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureResourceManagerClient.groovy [hunk 3]
```diff
@@ -264,6 +267,22 @@ class AzureResourceManagerClient extends AzureBaseClient {
...
+      @Override
+      Response intercept(Interceptor.Chain chain) throws IOException {
+        Request.Builder builder = chain.request().newBuilder()
+        def oldHeaderValue = chain.request().header("User-Agent")
+        if (oldHeaderValue) {
+          builder.header("User-Agent", "AzureSpinnaker ${oldHeaderValue}")
+        } else {
+          builder.header("User-Agent", "AzureSpinnaker")
+        }
+        return chain.proceed(builder.build())
```
Comment (HuanhuanSunMSFT) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureResourceManagerClient.groovy]: Same here.

**Review by HuanhuanSunMSFT — APPROVED** [review]

**Comment by scotmoor** [comment]
What about internal v. external subscriptions/users? Do we need to have a way to filter out internal users?

**Pull request closed without merging**